lib/purescript-cst/src/Language/PureScript/CST/Types.hs

```diff
@@ -76,16 +90,26 @@ data Token
   | TokInt !Text !Integer
   | TokNumber !Text !Double
   | TokLayoutStart
+    -- ^ Indicates the start of a new indentation block.
+    -- All expressions must be indented at least to here to be considered part of the block.
   | TokLayoutSep
+    -- ^ Indicates a separator between two entities within the same indentation block.
   | TokLayoutEnd
+    -- ^ Indicates the end of an indentation block.
+    -- All expressions after this one will be part of the previous block.
```

Comment:
I've tweaked your third bullet point slightly and used that as the doc comments for them. Rather than stating that for each data constructor, I've moved it to the type's docs.